static_frame/core/index_datetime.py

```diff
@@ -55,21 +56,20 @@
 #-------------------------------------------------------------------------------
 # Specialized index for dates
 
-class IndexDatetime(Index):
+class IndexDatetime(Index[np.datetime64]):
     '''
     Derivation of Index to support Datetime operations. Derived classes must define _DTYPE.
     '''
 
     STATIC = True
-    _DTYPE: DtypeAny # define in derived class
+    _DTYPE: TDtypeAny # define in derived class
```

Comment:
Wouldn't this be `_DTYPE: np.dtype[np.datetime64]`? All `IndexDatetime` subclasses need to have this dtype family

Reply:
Yes, good improvement!

Reply:
Yes, good find!